src/schema/v2/artwork/__tests__/artwork.test.js

```diff
@@ -3213,6 +3213,237 @@ describe("Artwork type", () => {
+      beforeEach(() => {
+        artwork.process_with_artsy_shipping_domestic = true
+      })
+
+      describe("when domestic_shipping_fee_cents and international_shipping_fee is null", () => {
+        beforeEach(() => {
+          artwork.domestic_shipping_fee_cents = null
+          artwork.international_shipping_fee_cents = null
+        })
+
+        it("returns false", () => {
+          return runQuery(query, context).then((data) => {
+            expect(data).toEqual({
+              artwork: {
+                isFixedShippingFeeOnly: false,
+              },
+            })
+          })
+        })
+      })
+
+      describe("when domestic_shipping_fee_cents is present", () => {
+        beforeEach(() => {
+          artwork.domestic_shipping_fee_cents = 100
+          artwork.international_shipping_fee_cents = null
+        })
+
+        it("returns false", () => {
+          return runQuery(query, context).then((data) => {
+            expect(data).toEqual({
+              artwork: {
+                isFixedShippingFeeOnly: false,
+              },
+            })
+          })
+        })
+      })
+
+      describe("when only international_shipping_fee_cents is present", () => {
+        beforeEach(() => {
+          artwork.domestic_shipping_fee_cents = null
+          artwork.international_shipping_fee_cents = 100
+        })
+
+        it("returns false", () => {
+          return runQuery(query, context).then((data) => {
+            expect(data).toEqual({
+              artwork: {
+                isFixedShippingFeeOnly: false,
+              },
+            })
+          })
+        })
+
+        describe("when only artsy international shipping is present", () => {
+          beforeEach(() => {
+            artwork.domestic_shipping_fee_cents = null
+            artwork.international_shipping_fee_cents = null
+            artwork.artsy_shipping_international = true
+          })
+
+          it("returns false", () => {
+            return runQuery(query, context).then((data) => {
+              expect(data).toEqual({
+                artwork: {
+                  isFixedShippingFeeOnly: false,
+                },
+              })
+            })
+          })
+        })
+
+        describe("when both domesting and international fee set to free", () => {
+          beforeEach(() => {
+            artwork.domestic_shipping_fee_cents = 0
+            artwork.international_shipping_fee_cents = 0
+          })
+
+          it("returns false", () => {
+            return runQuery(query, context).then((data) => {
+              expect(data).toEqual({
+                artwork: {
+                  isFixedShippingFeeOnly: false,
+                },
+              })
+            })
+          })
+        })
+      })
+    })
+
+    describe("when artsy domestic shipping disabled", () => {
+      beforeEach(() => {
+        artwork.process_with_artsy_shipping_domestic = false
+        artwork.artsy_shipping_international = false
```

Comment:
Are we covering all cases of overlap in the presence of `process_with_artsy_shipping_domestic` and `domestic_shipping_fee_cents`?

Reply:
I tried to think of all cases. But not sure what is the best way to organize the spec. Followed the pattern from the `onlyShipsDomestically`. Happy to add if I'm missing some.